File: Java/2501-2550/2540. Minimum Common Value.java
// https://leetcode.com/problems/minimum-common-value/description/

class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        int first = 0, second = 0; // Two-Pointer

        while(first < nums1.length && second < nums2.length){
            if(nums1[first] == nums2[second])
                return nums1[first]; // common value
            else if(nums1[first] < nums2[second])
                first++;
            else
                second++;
        }

        return -1; // No Common value found
    }
}

File: Python/2501-2550/2540. Minimum Common Value.py
# https://leetcode.com/problems/minimum-common-value/description/

class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        first, second = 0, 0 # Two-Pointer

        while first < len(nums1) and second < len(nums2) :
            if nums1[first] == nums2[second]:
                return nums1[first] # common value
            elif nums1[first] < nums2[second]:
                first += 1
            else:
                second += 1

        return -1 # No Common value found
